rumqttc/src/client.rs

```diff
@@ -367,23 +367,24 @@ impl Connection {
         let runtime = self.runtime.take().unwrap();
         Iter {
             connection: self,
-            runtime,
+            runtime: Some(runtime),
         }
     }
 }
 
 /// Iterator which polls the `EventLoop` for connection progress
 pub struct Iter<'a> {
     connection: &'a mut Connection,
-    runtime: runtime::Runtime,
+    runtime: Option<runtime::Runtime>,
```

Comment:
Is it possible to use runtime by reference (like connection) instead of wrapping it with an option and unwrapping everytime?

Reply:
I think the bigger question is do we want to allow consumer having multiple `Iter` at the same time?

If yes then we can just take a `&mut Connection` in the `Iter` and remove runtime completely. And run the executor using runtime in `Connection` itself. 

If no then we can probably wrap `Connection` inside a Inner type and use a singleton like pattern to get a `Iter`.

Reply:
I believe what you mean is an `Arc<Murex<Connection>>` as `&mut` is always exclusively owned.